File: Backend/Core/RawItem.cs
namespace Core;

public class RawItem : BaseEntity
{
    public string Name { get; set; }
    public string RawIconURL { get; set; }
    public double RawPrice { get; set; }
    public double OldPrice { get; set; }
    public bool IsOnSale { get; set; }
    public string RawItemURL { get; set; }
    public RawCategory RawCategory { get; set; }
    public long RawCategoryId { get; set; }
    public string Description { get; set; }
    public IList<Specification> Specifications { get; set; }
}

File: GoMicroservices/CoreStructs/RawItem.go
package CoreStructs

const RawItemTableName = "\"RawItems\""
const RawItemInsertQuery = "INSERT INTO " + RawItemTableName + " " +
	"(\"Id\", \"Name\", \"RawPrice\", \"OldPrice\", \"IsOnSale\", \"RawItemURL\", \"RawCategoryId\", \"Description\", \"RawIconURL\") VALUES " +
	"(:Id, :Name, :RawPrice, :OldPrice, :IsOnSale, :RawItemURL, :RawCategoryId, :Description, :RawIconURL)"

type RawItem struct {
	Id            int64   `db:"Id"`
	Name          string  `db:"Name"`
	RawPrice      float64 `db:"RawPrice"`
	OldPrice      float64 `db:"OldPrice"`
	IsOnSale      bool    `db:"IsOnSale"`
	RawItemURL    string  `db:"RawItemURL"`
	RawCategoryId int64   `db:"RawCategoryId"`
	Description   string  `db:"Description"`
	RawIconURL    string  `db:"RawIconURL"`
}
